rednotebook/util/markup.py

```diff
@@ -226,6 +226,24 @@ def _get_config(target, options):
         # {{red text|color:red}} -> <span style="color:red">red text</span>
         config['postproc'].append([COLOR_ESCAPED, r'<span style="color:\2">\1</span>'])
 
+        # Entry references
+        if is_exporting_to_file:
+            # When exporting we need to remove entry reference links because
+            # they are only valid within our app.
+            config['preproc'].append([r'\[(?P<name>.+)\s+(?P<date>\d{4}-\d{2}-\d{2})\s*\]',
+                                      r'\g<name> (\g<date>)'])
+        else:
+            # txt2tag will generate links to the named entry references because share common bracket notation
+            # used by the URIs. Here we just need to add our internal schema to make it a proper URI.
+            config['preproc'].append([r'\[(?P<name>.+)\s+(?P<date>\d{4}-\d{2}-\d{2})\s*\]',
+                                      r'[\g<name> ' + urls.INTERNAL_URI_SCHEMA + r':\g<date>]'])
+
+            # Stand alone dates are converted into named references where the date itself is being
+            # used as a name. For example:
+            # "Today is 2019-10-20" will be converted into "Today is [2019-10-20 notebook:2019-10-20]"
+            config['preproc'].append([r'(?<!:|\[)(?P<date>\d{4}-\d{2}-\d{2})',
+                                      r'[\g<date> ' + urls.INTERNAL_URI_SCHEMA + r':\g<date>]'])
```

Comment:
If we move this preproc before the other one, can we simplify the code a bit by removing INTERNAL_URI_SCHEMA here?

Reply:
I have decided to get rid of INTERNAL_URI_SCHEMA altogether - fragment links are a bit easier to handle (they are formed by just prepending `#` character) and will fit nicely with links in HTML exports I am currently working on. There will be no string concatenation so it will look better too :)
Next pull-request will merge code from `if is_exporting_to_file:` branches.